src/groupeddataframe/grouping.jl

```diff
@@ -179,6 +206,43 @@ end
 Base.getindex(gd::GroupedDataFrame, idxs::Colon) =
     GroupedDataFrame(gd.parent, gd.cols, gd.groups, gd.idx, gd.starts, gd.ends)
 
+function Base.keys(gd::GroupedDataFrame)
+    colnames = Tuple(_names(gd)[gd.cols])
```

Comment:
maybe sipler is `_names(gd.values)`? Same below

Reply:
Ah - and now I realize that if we materialize `values` then we run into the problem when `parent(gd)` columns are renamed. So probably it is safer not to store it, but compute dynamically (this adds a small overhead, but I assume that this function will not be used in performance-critical loops).

Reply:
or we should store only a vector of tuples of values (as group values should not change after grouped data frame is created).

Reply:
Maybe just store the values as a tuple of column vectors? That seems to be the most direct way to create a DataFrame, and it can get the names from the parent each time it's called.

Reply:
sounds OK

Reply:
Maybe we should return a custom iterator creating named tuples on the fly, rather than allocating a vector (that could just be a generator)? Also, better not store a `gd.values`, as it's going to add some overhead to all `groupby` operations even when it's not needed.

I was also thinking that iterator could itself return a special named tuple-like object which would internally keep track of the group number, so that we don't need to recompute if on indexing (which is terribly inefficient). What do you think?

Reply:
Thinking of it now - I think we can be lazy here. The benefit of materializing is if we did many iterations over a `GroupedDataFrame` after it is created, which is unlikely. Alternatively (but I think this is not needed given the discussion), we could do something similar that we do for `SubIndex` - materialize it only if needed (i.e. create an empty store and populate it only if iteration over group vales is required by a user).

Reply:
Now stores `.values` as vector of unnamed tuples, `keys()` gets parent column names when called. Added test for parent `DataFrame` column names being changed after creation.

Reply:
I unresolved it as I do not see these changes.

However (assuming that you have made changes but have not pushed them yet), what (I guess) @nalimilan and (for sure) I meant it that `values` should not be populated until it is being read for the first time.

Reply:
OK - I am resolving it here and I left a new comment in the code review section.